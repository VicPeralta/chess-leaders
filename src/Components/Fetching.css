.fetching {
  margin-top: 40px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  font-family: Lato, Arial, Helvetica, sans-serif;
}

.fetching p {
  font-size: 1.3rem;
  margin: 10px;
}
