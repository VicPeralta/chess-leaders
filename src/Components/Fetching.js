const Fetching = () => (
  <div>
    <h2>Fetching data from chess.com</h2>
    <p>Please waiter</p>
    <p>Spinner here</p>
  </div>
);

export default Fetching;
