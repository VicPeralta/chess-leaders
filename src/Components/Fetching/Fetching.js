import ProgressCircle from '../ProgressCircle/ProgressCircle';
import './Fetching.css';

const Fetching = () => (
  <div className="fetching">
    <p>Retrieving data </p>
    <p>from chess.com</p>
    <p>Please wait...</p>
    <ProgressCircle />
  </div>
);

export default Fetching;
