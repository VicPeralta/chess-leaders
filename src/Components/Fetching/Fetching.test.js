import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Fetching from './Fetching';

describe('Fetching Component testing', () => {
  test('Fetching rendering', () => {
    render(<Fetching />);
    const text = screen.getByText(/Retrieving data/i);
    expect(text).toBeInTheDocument();
  });
  test('Fetching Snapshot', () => {
    const fetching = renderer.create(<Fetching />);
    expect(fetching).toMatchSnapshot();
  });
});
