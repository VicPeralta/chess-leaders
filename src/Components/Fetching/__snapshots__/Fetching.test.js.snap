// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fetching Component testing Fetching Snapshot 1`] = `
<div
  className="fetching"
>
  <p>
    Retrieving data 
  </p>
  <p>
    from chess.com
  </p>
  <p>
    Please wait...
  </p>
  <div
    className="circle-wrap"
    data-testid="circle-element"
  >
    <div
      className="circle"
    >
      <div
        className="mask full"
      >
        <div
          className="fill"
        />
      </div>
      <div
        className="mask half"
      >
        <div
          className="fill"
        />
      </div>
      <div
        className="inside-circle"
      />
    </div>
  </div>
</div>
`;
